validation-update.yaml

```diff
@@ -0,0 +1,147 @@
+---
+- name: Run powershell script
+  hosts: windows
+  gather_facts: true
+
+  tasks:
+    - block:
+        - name: 1.Save the result from diag
+          win_command: dcdiag
+          ignore_errors: yes
+          register: dcdiag_output
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ dcdiag_output.stderr_lines }}"
+            verbosity: 2
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: 2.Save the result from Replication Status
+          win_shell: powershell.exe Get-Date
+          register: repl_out
+          ignore_errors: yes
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: repl_out
+            verbosity: 2
+
+        - name: 3.Save the result fron nslookup
+          win_command: nslookup 8.8.8.8
+          ignore_errors: yes
+          register: nslookup_test_1
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ nslookup_test_1 }}"
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: "nslookup Test 2."
+          ansible.windows.win_shell: nslookup pgcw001dc01
+          become: yes
+          become_user: Administrator
+          register: nslookup_test_2
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ nslookup_test_2 }}"
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: 4.Save the result from Active leases
```

Comment:
is there any logic is when we run the command and it fails , what next

Reply:
there is no such logic as like store output when command fails or passes. we run command and then either show full logs which inlcude stdout and stderr logs or only show stdout_lines or stderr_lines.

we can add logic in ansible to define when a command will fail. i.e lets say if command fails it outputs a string in log. we can add checks for that. best is to record all the logs and then do some logic on those logs to get final status of pass or fail of that command. In most cases ansible itself will mark a job pass or fail based on returned output i.e rc status